components/resources/quizzes/shared/StartQuiz.tsx

```diff
@@ -22,19 +42,27 @@ const StartQuiz = ({ onNextClick, body, title }: StartQuizProps) => {
           <input
             type="text"
             name="name"
+            value={userInput.name}
+            onChange={handleInputChange}
             className="w-4/5 ml-8 border border-gray-500 rounded-lg shadow appearance-none"
           ></input>
           <h3 className="ml-8">Email Address</h3>{" "}
           <input
             type="email"
             name="email"
+            value={userInput.email}
+            onChange={handleInputChange}
             className="w-4/5 ml-8 border border-gray-500 rounded-lg shadow appearance-none"
           ></input>
         </div>
         <Button
           backgroundColor="yellow"
           label="Start Quiz"
-          onClick={onNextClick}
+          disabled={userInput.name.length == 0 && userInput.email.length == 0}
+          onClick={() => {
+            onNextClick();
+            handleUserInputMutations(userInput);
```

Comment:
this component seems to get more complicated just because you want to call this function here. What if you moved this function inside of of the onNextClick function that is passed in from the parent?

Reply:
Done!